src/app/drive/components/FileViewer/utils/fileViewerWrapperUtils.ts

```diff
@@ -0,0 +1,220 @@
+    onShowDetailsButtonClicked,
+    onLinkSettingsButtonClicked,
+    onMoveToTrashButtonClicked,
+    onRenameItemButtonClicked,
+    onRestoreItemButtonClicked,
+    onDeletePermanentlyButtonClicked,
+  } = driveItemActions;
+
+  const driveActionsMenu = (): ListItemMenu<DriveItemData> => {
+    if (isSharedItem) {
+      return contextMenuDriveItemShared({
+        copyLink: onCopyLinkButtonClicked,
+        openShareAccessSettings: onLinkSettingsButtonClicked,
+        showDetails: onShowDetailsButtonClicked,
+        renameItem: onRenameItemButtonClicked,
+        moveItem: onMoveItemButtonClicked,
+        downloadItem: onDownloadItemButtonClicked,
+        moveToTrash: () => {
+          onMoveToTrashButtonClicked();
+          onClose();
+        },
+      });
+    } else {
+      return contextMenuDriveNotSharedLink({
+        shareLink: onLinkSettingsButtonClicked,
+        getLink: onCopyLinkButtonClicked,
+        renameItem: onRenameItemButtonClicked,
+        showDetails: onShowDetailsButtonClicked,
+        moveItem: onMoveItemButtonClicked,
+        downloadItem: onDownloadItemButtonClicked,
+        moveToTrash: () => {
+          onMoveToTrashButtonClicked();
+          onClose();
+        },
+      });
+    }
+  };
+
+  const recentsActionsMenu = (): ListItemMenu<DriveItemData> => {
+    return contextMenuDriveNotSharedLink({
+      shareLink: onLinkSettingsButtonClicked,
+      getLink: onCopyLinkButtonClicked,
+      renameItem: onRenameItemButtonClicked,
+      moveItem: onMoveItemButtonClicked,
+      showDetails: onShowDetailsButtonClicked,
+      downloadItem: onDownloadItemButtonClicked,
+      moveToTrash: () => {
+        onMoveToTrashButtonClicked();
+        onClose();
+      },
+    });
+  };
+
+  const sharedActionsMenu = (): ListItemMenu<AdvancedSharedItem> => {
+    return contextMenuDriveItemSharedAFS({
+      openShareAccessSettings: isOwner ? onLinkSettingsButtonClicked : undefined,
+      copyLink: onCopyLinkButtonClicked,
+      showDetails: onShowDetailsButtonClicked,
+      renameItem: !isCurrentUserViewer() ? onRenameItemButtonClicked : undefined,
+      moveItem: isOwner ? onMoveItemButtonClicked : undefined,
+      downloadItem: onDownloadItemButtonClicked,
+      moveToTrash: isOwner ? onShowStopSharingDialog : undefined,
+    });
+  };
+
+  const trashActionsMenu = (): ListItemMenu<DriveItemData> => {
+    return contextMenuTrashItems({
+      restoreItem: onRestoreItemButtonClicked,
+      deletePermanently: onDeletePermanentlyButtonClicked,
+    });
+  };
+
+  const contextMenuActions = () => {
+    if (isRecentsView) {
+      return recentsActionsMenu();
+    } else if (isSharedView) {
+      return sharedActionsMenu();
+    } else if (isTrashView) {
+      return trashActionsMenu();
+    } else {
+      return driveActionsMenu();
+    }
+  };
+
+  return contextMenuActions();
+};
+
+export function getFileContentManager(currentFile, downloadFile, handleFileThumbnail) {
```

Comment:
nit: Personally I prefer to export all the functions at the end of the file, it is easier to see what this file exports